sklearn/manifold/spectral_embedding_.py

```diff
@@ -263,7 +264,9 @@ def spectral_embedding(adjacency, n_components=8, eigen_solver=None,
         # problem.
         if not sparse.issparse(laplacian):
             warnings.warn("AMG works better for sparse matrices")
-        laplacian = laplacian.astype(np.float)  # lobpcg needs native floats
+        # lobpcg needs native floats
+        laplacian = check_array(laplacian, dtype=np.float, 
+                                accept_sparse=True, copy=False)
```

Comment:
We should use `np.float64` to make it explicit that that we want platform independent behavior with double precision floats.